### PR TITLE
Hotfix for vulnerabilities card with the number of vulnerabilities

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
- small fixes to fix The 405 Method Not Allowed error

image

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/table/filters/VulnerabilitiesFiltersRow.kt [hunk 1]
```diff
@@ -129,9 +129,10 @@ val vulnerabilitiesFiltersRow: FC<VulnerabilitiesFiltersProps> = FC { props ->
                 buttonBuilder(faSearch, classes = "btn mr-1", isOutline = props.filter == filter, style = "secondary") {
                     props.onChangeFilter(filter)
                 }
-                buttonBuilder(faWindowClose, classes = "btn mr-1", isOutline = true, style = "secondary") {
+                buttonBuilder(faWindowClose, classes = "btn mr-1", title = "Drop filters", isOutline = true, style = "secondary") {
                     props.onChangeFilter(null)
-                    setFilter { props.filter }
+                    // need to drop all tags
+                    setFilter { props.filter.copy(tags = emptySet()) }
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/table/filters/VulnerabilitiesFiltersRow.kt:135]: @sanyavertolet I decided to drop all tags when we drop filters, previously it was not very obvious. Drop means that we need to drop all filters, including tags

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 2]
```diff
@@ -57,23 +57,17 @@ class VulnerabilityController(
         @RequestBody filters: VulnerabilityFilter,
         authentication: Authentication?,
     ): Mono<VulnerabilityDtoList> = blockingToMono {
-        if (authentication?.hasRole(Role.SUPER_ADMIN) != true && filters.status != VulnerabilityStatus.APPROVED && !filters.isOwner) {
+        if (
+            // if user is not authenticated, he will have authentication = null and will not get other's submitted vulnerabilities
+            filters.status != VulnerabilityStatus.APPROVED && authentication?.name != filters.authorName &&
+                    // only if user is NOT admin, if admin - everything is fine
+                    authentication?.hasRole(Role.SUPER_ADMIN) == false
```
Comment (sanyavertolet) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:64]: Maybe a bit more readable way is `authentication?.hasRole(Role.SUPER_ADMIN) != true` as thus we can easily see that when `authentication?.hasRole(Role.SUPER_ADMIN) == null` everything is fine (just a point with readability)

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/table/filters/VulnerabilitiesFiltersRow.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/table/filters/VulnerabilitiesFiltersRow.kt:135]: oh ok

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityCollectionView.kt [hunk 3]
```diff
@@ -168,23 +167,25 @@ val vulnerabilityCollectionView: FC<VulnerabilityCollectionViewProps> = FC { pro
...
+                                    }
                                 }
                             }
 
                             vulnerabilityListTable {
                                 filters = vulnerabilityFilters
                                 getData = { _, _ ->
                                     post(
-                                        url = "$apiUrl/vulnerabilities/${props.currentUserInfo?.globalRole?.let { "by-filters" } ?: "get-all-public"}",
+                                        url = "$apiUrl/vulnerabilities/by-filters",
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityCollectionView.kt:188]: finally it is one endpoint!

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/VulnerabilityWelcomeView.kt [hunk 4]
```diff
@@ -109,25 +120,25 @@ val vulnWelcomeView: FC<WelcomeProps> = FC { props ->
...
 fun ChildrenBuilder.stats(vulnerabilities: List<VulnerabilityDto>) {
     div {
-        className = ClassName("card mt-3")
+        className = ClassName("card border border-primary rounded rounded-pill col mt-4 justify-content-center")
+        style = jso {
+            height = 15.rem
+        }
         div {
-            className = ClassName("mt-5 mb-5")
-            style = jso {
```
[the remark this answers is not in the document]
Comment (JustinB1eber) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/VulnerabilityWelcomeView.kt:115]: got it!

**Review by JustinB1eber — APPROVED** [review]

**Review by Cheshiriks — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 2 above)
Comment (akuleshov7) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:64]: I thought in this way, but here is the logic is that you don't have SUPER ADMIN role (your role superadmin is false, like a flag)

**Pull request merged**